trifle/program/src/processor.rs

```diff
@@ -832,3 +837,49 @@ fn add_tokens_constraint_to_escrow_constraint_model(
         constraint,
     )
 }
+
+fn remove_constraint_from_escrow_constraint_model(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    args: RemoveConstraintFromEscrowConstraintModelArgs,
+) -> ProgramResult {
+    let mut accounts_iter = accounts.iter();
+    let escrow_constraint_model_info = next_account_info(&mut accounts_iter)?;
+    let payer_info = next_account_info(&mut accounts_iter)?;
+    let update_authority_info = next_account_info(&mut accounts_iter)?;
+    let system_program_info = next_account_info(&mut accounts_iter)?;
+
+    assert_signer(payer_info)?;
+    assert_signer(update_authority_info)?;
+    assert_owned_by(escrow_constraint_model_info, program_id)?;
+    // assert update authority matches ecm update authority;
+    let mut escrow_constraint_model =
+        EscrowConstraintModel::try_from_slice(&escrow_constraint_model_info.data.borrow())?;
+
+    if escrow_constraint_model.update_authority != *update_authority_info.key {
+        return Err(TrifleError::InvalidUpdateAuthority.into());
+    }
+
+    // remove the constraint by key.
+    escrow_constraint_model
+        .constraints
+        .remove(&args.constraint_name);
+
+    let serialized_data = escrow_constraint_model.try_to_vec().unwrap();
```

Comment:
I have learned Sam doesn't approve anything with unwraps so this should be changed to
```
let serialized_data = escrow_constraint_model
        .try_to_vec()
        .map_err(|_| MetadataError::BorshSerializationError)?;
```

Reply:
Or similar error.